forge-gui/src/main/java/forge/gamemodes/net/server/FServerManager.java

```diff
@@ -75,6 +83,35 @@ public static FServerManager getInstance() {
     }
 
     public void startServer(final int port) {
+        this.port = port;
+        String UPnPOption = FModel.getNetPreferences().getPref(ForgeNetPreferences.FNetPref.UPnP);
+        boolean startUPnP;
+        if(UPnPOption.equalsIgnoreCase("ASK")) {
+            switch (SOptionPane.showOptionDialog(localizer.getMessageorUseDefault("lblUPnPQuestion", String.format("Attempt to open port %d automatically?", port), port),
+                    localizer.getMessageorUseDefault("lblUPnPTitle", "UPnP option"),
+                    null,
+                    ImmutableList.of(localizer.getMessageorUseDefault("lblUPnPJustOnce", "Just Once"),
+                            localizer.getMessageorUseDefault("lblUPnPNotNow", "Not Now"),
+                            localizer.getMessageorUseDefault("lblUPnPAlways", "Always"),
+                            localizer.getMessageorUseDefault("lblUPnPNever", "Never")), 0)) {
+                case 2:
+                    FModel.getNetPreferences().setPref(ForgeNetPreferences.FNetPref.UPnP, "ALWAYS");
+                    FModel.getNetPreferences().save();
+                case 0:
+                    //case 2 falls in here
+                    startUPnP = true;
+                    break;
+
+                case 3:
+                    FModel.getNetPreferences().setPref(ForgeNetPreferences.FNetPref.UPnP, "NEVER");
+                    FModel.getNetPreferences().save();
+                default:
+                    //case 1 defaults to here
+                    startUPnP = false;
+                    break;
+            }
+        } else startUPnP = UPnPOption.equalsIgnoreCase("ALWAYS");
```

Comment:
Can we wrap this in braces? This same line else is vary jarring within this code style.